slmsuite/hardware/cameras/simulated.py

```diff
@@ -0,0 +1,219 @@
+        if f_eff is None:
+            self.f_eff = f_min
+            print("Setting f_eff = f_min = %1.2f pix/rad to place camera \
+within accessible SLM k-space."%(self.f_eff))
+        elif f_eff < f_min:
+            raise RuntimeError("Camera extends beyond SLM's accessible Fourier space!")
+        else:
+            self.f_eff = f_eff
+        
+        # Fourier space must be sufficiently padded to resolve the camera pixels.
+        # Future: account for anisotropic x,y resolution when non-square pixel is rotated.
+        self.shape_padded = Hologram.calculate_padded_shape(slm, precision=1/self.f_eff)
+        self._hologram = Hologram(
+            self.shape_padded,
+            amp=self._slm.amp_profile,
+            phase=self._slm.phase + self._slm.phase_offset,
+            slm_shape=self._slm,
+        )
+        print("Padded SLM k-space shape set to (%d,%d) to achieve \
+required imaging resolution."%(self.shape_padded[1], self.shape_padded[0]))
+
+    def flush(self):
+        """
+        See :meth:`.Camera.flush`.
+        """
+        return
+
+    def set_exposure(self, exposure):
+        """
+        Set the simulated exposure (i.e. digital gain).
+
+        Parameters
+        ----------
+        exposure : float
+            Digital gain.
+        """
+        self.exposure = exposure
+
+    def get_exposure(self):
+        """
+        Get the simulated exposure (i.e. digital gain).
+        """
+        return self.exposure
+
+    def get_image(self, plot=False):
+        """
+        See :meth:`.Camera.get_image`. Computes and samples the affine-transformed SLM far-field.
+
+        Parameters
+        ----------
+        plot : bool
+            Whether to plot the output.
+
+        Returns
+        -------
+        numpy.ndarray
+            Array of shape :attr:`shape`
+        """
+
+        # Update phase; calculate the far-field (keep on GPU if using cupy for follow-on interp)
+        self._hologram.reset_phase(self._slm.phase + self._slm.phase_offset)
+        ff = self._hologram.extract_farfield(get = True if (mp==np) else False)
+
+        # Use map_coordinates for fastest interpolation; but need to reshape pixel dimensions
+        # to account for additional padding.
+        if self._interpolate:
+            img = map_coordinates(mp.abs(ff)**2,mp.array([
+                                            self.shape_padded[0]/(self.f_eff/self._slm.dy)*
+                                            self.ygrid+self.shape_padded[0]/2,
+                                            self.shape_padded[1]/(self.f_eff/self._slm.dx)*
+                                            self.xgrid+self.shape_padded[1]/2]),
+                                            order=0)
+        else:
+            img = mp.abs(ff)**2
+            img = toolbox.unpad(img,self.shape)
+        if mp != np:
+            img = img.get()
+        img = self.exposure * img
+
+        if plot:
```

Comment:
Plotting should probably either be removed or added to the superclass.

Reply:
Moved to a `staticmethod` under superclass. @ichristen Request your re-review.